File: src/test/java/ru/jerael/booktracker/backend/data/mapper/GenreDataMapperTest.java
package ru.jerael.booktracker.backend.data.mapper;

import org.junit.jupiter.api.Test;
import ru.jerael.booktracker.backend.data.db.entity.GenreEntity;
import ru.jerael.booktracker.backend.domain.model.Genre;
import static org.junit.jupiter.api.Assertions.assertEquals;

class GenreDataMapperTest {
    private final GenreDataMapper genreDataMapper = new GenreDataMapper();

    @Test
    void toDomain() {
        GenreEntity entity = new GenreEntity();
        entity.setId(1);
        entity.setName("adventure");

        Genre genre = genreDataMapper.toDomain(entity);

        assertEquals(1, genre.id());
        assertEquals("adventure", genre.name());
    }

    @Test
    void toEntity() {
        Genre genre = new Genre(1, "adventure");

        GenreEntity entity = genreDataMapper.toEntity(genre);

        assertEquals(1, entity.getId());
        assertEquals("adventure", entity.getName());
    }
}

File: src/test/kotlin/ru/jerael/booktracker/backend/data/mapper/GenreDataMapperTest.kt
package ru.jerael.booktracker.backend.data.mapper

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import ru.jerael.booktracker.backend.factory.genre.GenreDomainFactory
import ru.jerael.booktracker.backend.factory.genre.GenreEntityFactory

class GenreDataMapperTest {
    private val mapper = GenreDataMapper()
    
    @Test
    fun `toDomain should map GenreEntity to Genre`() {
        val entity = GenreEntityFactory.createGenreEntity()
        
        val genre = mapper.toDomain(entity)
        
        with(genre) {
            assertEquals(entity.id, id)
            assertEquals(entity.name, name)
        }
    }
    
    @Test
    fun `toEntity should map Genre to GenreEntity`() {
        val genre = GenreDomainFactory.createGenre()
        
        val entity = mapper.toEntity(genre)
        
        with(entity) {
            assertEquals(genre.id, id)
            assertEquals(genre.name, name)
        }
    }
}